lib/auth/resetpasswordtoken.go

```diff
@@ -112,6 +112,10 @@ func (s *Server) CreateResetPasswordToken(ctx context.Context, req CreateResetPa
 		return nil, trace.Wrap(err)
 	}
 
+	if err := s.resetMFA(ctx, req.Name); err != nil {
```

Comment:
Hmm, I'm curious about a scenario when somebody requests a password reset link but then changes their mind - maybe recalls the password, or wrongfully generated a link for a different user. That user's devices will be gone. Do you think it might make sense to reset them upon password reset form submission instead of upon token generation?

Reply:
This follows the precedent of passwords (a few lines above).
We replace user's password with a random one to prevent them from logging in after reset was initiated.

I don't think you can go back after password reset was initiated.

Reply:
Gotcha. Guess it makes sense from the security perspective.